### ISSUE TITLE
[Bug]: `--no-defaults` flag is ignore on `yarn config` when the `--json` flag is present 

### ISSUE THREAD

**Issue by chrisbodhi** [yarnpkg/berry#6341]
### Self-service

- [ ] I'd be willing to implement a fix

### Describe the bug

Running `yarn config --no-defaults --json` returns in a JSON format (expected) all of the config options (unexpected). I expected the `yarn config` command, with both flags, to return only the non-default settings in a JSON format.

### To reproduce

`yarn config --json --no-defaults`

### Environment

```shell
System:
    OS: macOS 14.4.1
    CPU: (10) arm64 Apple M1 Pro
  Binaries:
    Node: 20.7.0 - /private/var/folders/lc/yqxr0r45775_2f5mfzyj9x2c0000gp/T/xfs-fb1e8fe6/node
    Yarn: 4.2.2 - /private/var/folders/lc/yqxr0r45775_2f5mfzyj9x2c0000gp/T/xfs-fb1e8fe6/yarn
    npm: 10.1.0 - ~/.nvm/versions/node/v20.7.0/bin/npm
```

**Comment by sparrowt** [comment]
Yes this is frustrating - this bug seems to occur because `this.noDefaults` is only checked [here](https://github.com/yarnpkg/berry/blob/8bfe2d545e986993e4450072bac8b1044e5ebed7/packages/plugin-essentials/sources/commands/config.ts#L106) which is in the `else` block of the `if (this.json) {` so it is not checked when emitting the JSON output.

**Comment by sparrowt** [comment]
I've proposed a fix here https://github.com/yarnpkg/berry/pull/6635
